pkg/config/config.go

```diff
@@ -0,0 +1,105 @@
+// Package config has specific primitives for loading application configurations.
+//
+// Primitives:
+// - 	Application should have struct for containing configuration. E.g. refer
+// 		trino-gateway/internal/config/config.go file.
+// - 	Application should have a directory holding default file and environment
+// 		specific file. E.g. refer trino-gateway/configs/* directory.
+//
+// Usage:
+// - 	E.g. NewDefaultConfig().Load("dev", &config), where config is a struct
+// 		where configuration gets unmarshalled into.
+package config
+
+import (
+	"os"
+	"path"
+	"runtime"
+	"strings"
+
+	"github.com/spf13/viper"
+)
+
+// Default options for configuration loading.
+const (
+	DefaultConfigType     = "toml"
+	DefaultConfigDir      = "./config"
+	DefaultConfigFileName = "default"
+	WorkDirEnv            = "WORKDIR"
+)
+
+// Options is config options.
+type Options struct {
+	configType            string
+	configPath            string
+	defaultConfigFileName string
+}
+
+// Config is a wrapper over a underlying config loader implementation.
+type Config struct {
+	opts  Options
+	viper *viper.Viper
+}
+
+// NewDefaultOptions returns default options.
+// DISCLAIMER: This function is a bit hacky
+// This function expects an env $WORKDIR to
+// be set and reads configs from $WORKDIR/configs.
+// If $WORKDIR is not set. It uses the absolute path wrt
+// the location of this file (config.go) to set configPath
+// to 2 levels up in viper (../../configs).
+// This function breaks if :
+// 1. $WORKDIR is set and configs dir not present in $WORKDIR
+// 2. $WORKDIR is not set and ../../configs is not present
+// 3. $WORKDIR is not set and runtime absolute path of configs
+// is different than build time path as runtime.Caller() evaluates
+// only at build time
+func NewDefaultOptions() Options {
+	var configPath string
+	workDir := os.Getenv(WorkDirEnv)
+	if workDir != "" {
+		configPath = path.Join(workDir, DefaultConfigDir)
+	} else {
+		_, thisFile, _, _ := runtime.Caller(1)
```

Comment:
Can we log that `$WORKDIR` is not set and that we are trying to read from default configs at path x/y/z?